tests/integration.test.ts

```diff
@@ -1,126 +1,258 @@
+    fromUser.adminPods = [addressTwo];
+    fromUser.save();
+
+    let userPod1 = new PodUser(addressOne + "-1");
+    userPod1.user = addressOne;
+    userPod1.pod = "1";
+    userPod1.save();
+
+    let userPod2 = new PodUser(addressOne + "-2");
+    userPod2.user = addressOne;
+    userPod2.pod = "2";
+    userPod2.save();
+
+    // Burn two tokens, i.e., transfer to addressZero
+    let transferBatch = generateTransferBatch(
+      addressOne,
+      addressOne,
+      addressZero,
+      [1, 2],
+      [1, 1]
+    );
+
+    assert.fieldEquals("PodUser", addressOne + "-1", "user", addressOne);
+    assert.fieldEquals("PodUser", addressOne + "-1", "pod", "1");
+    handleTransferBatch(transferBatch);
+    assert.notInStore("PodUser", addressZero + "-1");
+    assert.notInStore("User", addressZero);
+    clearStore();
+  });
+
+  test("A user can exist in multiple pods", () => {
+    let transferBatchEvent = generateTransferBatch(
+      addressOne,
+      addressZero,
+      addressOne,
+      [1, 2],
+      [1, 1]
+    );
+    handleTransferBatch(transferBatchEvent);
+
+    assert.fieldEquals("User", addressOne, "id", addressOne);
+    assert.fieldEquals("PodUser", addressOne + "-1", "user", addressOne);
+    assert.fieldEquals("PodUser", addressOne + "-2", "user", addressOne);
+
+    clearStore();
+  });
 });
+//TODO test deregisterevent for different controllers
+// abstract handleDeregisterPod and take a version number
 
-test('DeregisterPod should delete the pod entity', () => {
+describe("DeregisterPod", () => {
+  test("DeregisterPod should delete the pod entity", () => {
     // Create a pod
     let transferSingleEvent = generateTransferSingle(
       addressOne,
       addressZero,
       addressOne,
       119,
-      1,
+      1
     );
     handleTransferSingle(transferSingleEvent);
 
-    assert.fieldEquals('Pod', '119', 'id', '119');
-    // Admin should always be null for a fresh pod on creation.
+    assert.fieldEquals("Pod", "119", "id", "119");
+    assert.fieldEquals("User", addressOne, "adminPods", "[]");
     // Updating should be handled by UpdatePodAdmin, not this function.
-    assert.fieldEquals('User', addressOne, 'id', addressOne);
-    assert.fieldEquals('PodUser', addressOne + '-119', 'user', addressOne);
-    assert.fieldEquals('PodUser', addressOne + '-119', 'pod', '119');
+    assert.fieldEquals("User", addressOne, "id", addressOne);
+    assert.fieldEquals("PodUser", addressOne + "-119", "user", addressOne);
+    assert.fieldEquals("PodUser", addressOne + "-119", "pod", "119");
 
-    let deregisterPodEvent = generateDeregisterPod(119);
+    let deregisterPodEvent = generateDeregisterPodV1_4(119);
     handleDeregisterPod(deregisterPodEvent);
 
     // Pod and PodUser should be deleted
-    assert.notInStore('Pod', '119');
-    assert.notInStore('PodUser', addressOne + '-119');
+    assert.notInStore("Pod", "119");
+    assert.notInStore("PodUser", addressOne + "-119");
```

Comment:
@jtlacci , we were wondering if we needed to remove old references to the `PodUser` from the user's `.pods` list.  When I added an assertion for this: 
```
    //check that we remove the PodUsers for the pod.
    assert.fieldEquals("User", addressOne, "pods", "[]");
```
it failed with 
```
No field named 'pods' on entity with type 'User' and id '0x1
```
I assume that's because the `user.pods` attributed is "derived" in the schema:
```
type User @entity {
  id: ID! # address
  pods: [PodUser!] @derivedFrom(field: "user")
  ...
}
```
So I'm not sure how we'd test this.

Reply:
can we try to test a user with multiple pods to make sure that only the deregistered one is getting removed?

Reply:
Okay, I don't think it's failing because it's derived, it's failing because we never set it 😬

Reply:
Added below